On Sun, Feb 22, 2004 at 05:22:00PM -0800, Andrew Morton wrote:
> ftp://ftp.kernel.org/pub/linux/kernel/people/akpm/patches/2.6/2.6.3/2.6.3-mm2/

Rusty's patch 'add-MODULE_VERSION-macro.patch' is missing a change that
results in modules_install installing all of the modules AND their
constituent source files into /lib/modules. The change, which did appear
in Makefile.modpost correctly, follows here.

--- scripts/Makefile.modinst~	2004-02-22 19:31:28.000000000 -0800
+++ scripts/Makefile.modinst	2004-02-22 19:31:49.000000000 -0800
@@ -9,7 +9,7 @@
 
 #
 
-__modules := $(shell cat /dev/null $(wildcard $(MODVERDIR)/*.mod))
+__modules := $(shell head -q -n1 $(wildcard $(MODVERDIR)/*.mod))
 modules := $(patsubst %.o,%.ko,$(wildcard $(__modules:.ko=.o)))
 
 ifneq ($(filter-out $(modules),$(__modules)),)
 
-- 
Joshua Kwan
On Sun, Feb 22, 2004 at 05:22:00PM -0800, Andrew Morton wrote:
> ftp://ftp.kernel.org/pub/linux/kernel/people/akpm/patches/2.6/2.6.3/2.6.3-mm3/

The change made to scripts/Makefile to rake in sumversion.o requires
this change to not break parallel builds:

--- scripts/Makefile~	2004-02-22 18:50:39.000000000 -0800
+++ scripts/Makefile	2004-02-22 18:51:06.000000000 -0800
@@ -24,7 +24,7 @@
 
 # dependencies on generated files need to be listed explicitly
 
-$(obj)/modpost.o $(obj)/file2alias.o: $(obj)/elfconfig.h
+$(addprefix $(obj)/,$(modpost-objs)): $(obj)/elfconfig.h
 
 quiet_cmd_elfconfig = MKELF   $@
       cmd_elfconfig = $(obj)/mk_elfconfig $(ARCH) < $< > $@

Seems they all depend on elfconfig.h being there, so made it a general
dependency.

-- 
Joshua Kwan
